rtloader/common/builtins/datadog_agent.c

```diff
@@ -927,3 +934,40 @@ static PyObject *obfuscate_mongodb_string(PyObject *self, PyObject *args, PyObje
     PyGILState_Release(gstate);
     return retval;
 }
+
+/*! \fn PyObject *emit_agent_telemetry(PyObject *self, PyObject *args, PyObject *kwargs)
+    \brief This function implements the `datadog_agent.emit_agent_telemetry` method, emitting agent elemetry
+    for the provided check, metric, and value.
+    \param self A PyObject* pointer to the `datadog_agent` module.
+    \param args A PyObject* pointer to a tuple containing the key to retrieve.
+    \param kwargs A PyObject* pointer to a map of key value pairs.
+    \return A PyObject* pointer to the value.
+
+    This function is callable as the `datadog_agent.emit_agent_telemetry` Python method and
+    uses the `cb_emit_agent_telemetry()` callback to emit the agent telemetry
+    with CGO. If the callback has not been set `None` will be returned.
+*/
+static PyObject *emit_agent_telemetry(PyObject *self, PyObject *args, PyObject *kwargs)
+{
+    // callback must be set
+    if (cb_emit_agent_telemetry == NULL) {
+        Py_RETURN_NONE;
+    }
+
+    PyGILState_STATE gstate = PyGILState_Ensure();
+
+    char *check = NULL;
+    char *metric = NULL;
+    float value;
+    char *type = NULL;
+    if (!PyArg_ParseTuple(args, "ssfs", &check, &metric, &value, &type)) {
+        PyGILState_Release(gstate);
+        return NULL;
```

Comment:
Question: (Not to be address on this PR) I noticed we have this pattern of returning `NULL` in the case of failing to parse the Object. Are we OK with silent failing? Would it be best to log something?

Reply:
The failure isn't silent, it is just handled under the hood:

> Returns true on success; on failure, it returns false and raises the appropriate exception.

https://docs.python.org/3/c-api/arg.html#c.PyArg_ParseTuple

Reply:
Thanks for the link @vickenty